MAINTAINERS.adoc

```diff
@@ -0,0 +1,129 @@
+.. `git clean -nxd` should be clean or close to it.  It should show nothing that should be checked into git and probably no stale build artifacts.
+.. It's a good idea to check that CI completed successfully on GitHub for the commit that you've got checked out.  This will ensure that we're publishing a release that builds from scratch, passes all tests on all platforms, and passes our style and lint checks.
+... Use `git log -1` to see what commit you're on.
+... Look up that commit on GitHub.  It will have a URL like `https://github.com/oxidecomputer/dropshot/commit/COMMIT_SHA` (with an actual `COMMIT_SHA`).  CI succeeded if there's a green check at the left of the commit message.  You can click the green check to be sure you're looking at the right thing.  It should say "All checks have passed".
+. Figure out the tag used for the most recent release.  The easiest is probably to view https://github.com/oxidecomputer/dropshot/releases[the recent releases].   We'll call this `PREV_RELEASE_TAG` in the next step.
+. Figure out what kind of update this is relative to the last one released.  https://github.com/sunng87/cargo-release/blob/master/docs/reference.md#bump-level[See `cargo release` docs for the available levels.]  We're usually doing a "patch", "minor", or "major" release.  We'll call this "RELEASE_KIND".  (According to the docs, you can also use a specific version number that you want to release here.)
+. Do the release.  If you're daring, just run:
++
+[source,text]
+----
+$ cargo release --prev-tag-name=PREV_RELEASE_TAG -vv RELEASE_KIND
+----
++
+If you want to be extra careful, you can do this in a dry-run form.  First, check that the log output here looks right (and especially that you got the version number that you expected!):
++
+[source,text]
+----
+$ cargo release --dry-run --prev-tag-name=PREV_RELEASE_TAG -vv RELEASE_KIND
+----
++
+Now, run `cargo release` for real (i.e. no `--dry-run`), but without publishing to crates.io or pushing the git tags:
++
+[source,text]
+----
+$ cargo release --skip-push --skip-publish --prev-tag-name=PREV_RELEASE_TAG -vv minor
+----
++
+Inspect the resulting commits and the new tags from above.  There should be four commits: one to each of "dropshot" and "dropshot_endpoint" that sets the version to the one you specified, and one that sets the version to the _next_ version.  For example, if the current version is 0.5.1 and you're publishing 0.5.2, the repo will start at 0.5.2-pre, you'll see one commit that sets it to 0.5.2, and then you'll see one that sets it to 0.5.3-pre.
++
+Then undo the above step:
++
+[source,text]
+----
+$ git reset --hard THE_COMMIT_YOU_STARTED_WITH
+$ git tag -d THE_NEW_TAG
+----
++
+and do the release again with the publish step:
++
+[source,text]
+----
+$ cargo release --skip-push --prev-tag-name=PREV_RELEASE_TAG -vv minor
+----
++
+At this point, the new crates should be published to crates.io.  Check and
+push the commits and the tags.
```

Comment:
To make sure I'm understanding correctly: The crate is published *before* the commits are pushed? Is this because we're creating tags and publishing, but explicitly passing `--skip-push`?

Is there a reason we're avoiding letting `cargo release` do this for us?

Reply:
That is correct.  The reason is what I mentioned above: I wanted to be able to check the tool's output as much as possible before committing to it.  Separating this allows one to check them before pushing.  (Yes, we may have already published, but that's why I also do a the non-publishing version first, too.)

The one-liner at the top of this section has `cargo release` do it for you.